compiler/src/iree/compiler/Codegen/Common/GPU/GPUTileAndConvertConvToMatmul.cpp

```diff
@@ -0,0 +1,190 @@
+  for (auto attr : op->getDiscardableAttrs()) {
+    if (attr.getName() != memoAttr) {
+      prunedAttributeList.push_back(attr);
+    }
+  }
+  return prunedAttributeList;
+}
+
+// Helper to remove unit filter loop dimensions from input map of convolution
+// operations so that they can become contractions.
+void static removeUnitExtentDimsfromMaps(linalg::LinalgOp linalgOp,
+                                         RewriterBase &rewriter) {
+  auto convDimsOrFailure = linalg::inferConvolutionDims(linalgOp);
+  if (failed(convDimsOrFailure)) {
+    return;
+  }
+  const mlir::linalg::ConvolutionDimensions &convDims = *convDimsOrFailure;
+  // We cant make strided convolutions into contractions directly so bail out.
+  if (llvm::any_of(convDims.strides,
+                   [](int64_t stride) { return stride != 1; })) {
+    return;
+  }
+  SmallVector<AffineMap> indexingMaps = linalgOp.getIndexingMapsArray();
+  if (indexingMaps.empty())
+    return;
+  AffineMap inputMap = indexingMaps[0];
+  AffineMap filterMap = indexingMaps[1];
+
+  // Check that all filter loop dimensions are unit.
+  Value filter = linalgOp.getDpsInputs()[1];
+  auto filterType = llvm::cast<ShapedType>(filter.getType());
+  ArrayRef<int64_t> filterShape = filterType.getShape();
+  for (auto filterLoop : convDims.filterLoop) {
+    std::optional<int64_t> maybeDim = filterMap.getResultPosition(
+        getAffineDimExpr(filterLoop, filterMap.getContext()));
+    if (!maybeDim || filterShape[maybeDim.value()] != 1) {
+      return;
+    }
+  }
+  // Remove the unit filter loop dims form input map.
+  SmallVector<AffineMap> newIndexingMaps;
+  SmallVector<AffineExpr> newExprs;
+  for (AffineExpr expr : inputMap.getResults()) {
+    AffineExpr newExpr = expr;
+    if (auto binaryExpr = llvm::dyn_cast<AffineBinaryOpExpr>(expr)) {
+      for (auto filterLoop : convDims.filterLoop) {
+        if (binaryExpr.getLHS().isFunctionOfDim(filterLoop)) {
+          newExpr = binaryExpr.getRHS();
+        } else if (binaryExpr.getRHS().isFunctionOfDim(filterLoop)) {
+          newExpr = binaryExpr.getLHS();
+        }
+      }
+    }
+    newExprs.push_back(newExpr);
+  }
+  newIndexingMaps.push_back(AffineMap::get(inputMap.getNumDims(), 0, newExprs,
+                                           rewriter.getContext()));
+  // No changes to the filter and output map.
+  newIndexingMaps.push_back(filterMap);
+  newIndexingMaps.push_back(indexingMaps[2]);
+  // Create the new contraction op and replace the old convolution op.
+  auto newOp = rewriter.create<linalg::GenericOp>(
+      linalgOp.getLoc(), linalgOp.getDpsInits().getType(),
+      linalgOp.getDpsInputs(), linalgOp.getDpsInits(), newIndexingMaps,
+      linalgOp.getIteratorTypesArray(), /*bodyBuild=*/nullptr,
+      getPrunedAttributeList(linalgOp));
+  rewriter.inlineRegionBefore(linalgOp->getRegion(0), newOp.getRegion(),
+                              newOp.getRegion().begin());
+  rewriter.replaceOp(linalgOp, newOp.getResults());
```

Comment:
optional nit: Maybe just:
```suggestion
  if (!isa<linalg::GenericOp>(linalgOp.getOperation())) {
    linalgOp = linalg::generalizeNamedOp(rewriter, linalgOp).value();
  }
  rewriter.modifyOpInPlace([&]() {
    linalgOp.setIndexingMapsAttr(rewriter.getAffineMapArrayAttr(newIndexingMaps));
  });
```

Reply:
Oh, actually, this might erase the lowering config when generalizing. Maybe it doesn't end up being much of a simplification after all.

Reply:
Yeah we did this when implementing the genralized conv to igemm conversion and unfortunately this is the only way to support both named and generic ops that wont drop the config.